applications/minotari_console_wallet/src/grpc/wallet_debouncer.rs

```diff
@@ -177,9 +185,15 @@ impl WalletDebouncer {
                                 TransactionEvent::DetectedTransactionConfirmed { .. } |
                                 TransactionEvent::TransactionMined { .. } |
                                 TransactionEvent::TransactionMinedUnconfirmed { .. } |
-                                TransactionEvent::TransactionImported(_)  |
-                                TransactionEvent::TransactionValidationStateChanged(..) => {
+                                TransactionEvent::TransactionImported(_)  => {
+                                    self.set_refresh_needed(true).await;
+                                },
+                                TransactionEvent::TransactionValidationStateChanged{faux, ..} => {
                                     self.set_refresh_needed(true).await;
+                                    if faux {
+                                        let mut intial = self.initial_validation_done.lock().await;
+                                        *intial = true;
+                                    }
```

Comment:
I would use the first `TransactionEvent::TransactionValidationCompleted(` to set 
```
 let mut intial = self.initial_validation_done.lock().await;
*intial = true;
```
if it is false, and then we do not have to set the `faux` flag. The `faux` flag in `TransactionEvent::TransactionValidationStateChanged` is not intuitive.

Reply:
thats not going to catch all cases, we the faux handling happens last of the tx validation, and we need to ensure this is run after scanning completed